routes/landfastice.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+import numpy as np
+import pandas as pd
+from flask import (
+    Blueprint,
+    render_template,
+    request,
+)
+
+# local imports
+from fetch_data import (
+    fetch_wcs_point_data,
+    get_dim_encodings,
+    build_csv_dicts,
+    write_csv,
+)
+from validate_request import (
+    validate_latlon,
+    project_latlon,
+)
+from validate_data import postprocess
+from . import routes
+from config import WEST_BBOX, EAST_BBOX
+
+landfastice_api = Blueprint("landfastice_api", __name__)
+landfastice_coverage_id = "landfast_sea_ice_extent"
+landfastice_encodings = asyncio.run(
+    get_dim_encodings(landfastice_coverage_id, scrape=("time", "gmlrgrid:coefficients"))
+)
+
+
+def generate_time_index(ice_season="all"):
+    """Generate a Pythonic time index for a single October-July ice season.
+
+    Arguments:
+        ice_season (str): Ice season string (YYYY) between 1996 and 2007 or 'all' for the entire time series.
+
+    Returns:
+        dt_range (pandas DatetimeIndex): a time index with daily frequency
+    """
+
+    if ice_season == "all":
+        timestamps = [x[1:-2] for x in landfastice_encodings["time"].split(" ")]
+        date_index = pd.DatetimeIndex(timestamps)
+    else:
+        start, end = landfastice_encodings["ice_season"][ice_season][0].split(", ")
+        dt_start = start.split("T", 1)[0]
+        dt_end = end.split("T", 1)[0]
+        date_index = pd.date_range(start=dt_start, end=dt_end)
+    return date_index
+
+
+def package_landfastice_data(landfastice_resp):
+    """Package landfast ice extent data into a nested JSON-like dict.
+
+    Arguments:
+        time_index (pandas DateRange object) -- a time index with daily frequency
+        landfastice_resp (list) -- the response from the WCS GetCoverage request
+
+    Returns:
+        di (dict) -- a dict where the key is a single date and the value is the landfast ice status (1 or 0)
+    """
+    time_index = generate_time_index("all")
+    di = {}
+    for t, x in zip(list(time_index), landfastice_resp):
+        di[t.date().strftime("%m-%d-%Y")] = x
+    return di
+
+
+def create_landfast_csv(data_pkg, lat=None, lon=None):
+    """Create CSV file with metadata string and location based filename.
+    Args:
+        data_pkg (dict): JSON-like object of data
+        lat: latitude for points or None for polygons
+        lon: longitude for points or None for polygons
+    Returns:
+        CSV response object
+    """
+    fieldnames = [
+        "date",
+        "value",
+    ]
+    csv_dicts = build_csv_dicts(
+        data_pkg,
+        fieldnames,
+    )
+    metadata = "# Landfast Ice Status: 0 indicates absence and 1 indicates presence.\n"
+    filename = "Landfast Ice Extent for " + lat + ", " + lon + ".csv"
+    return write_csv(csv_dicts, fieldnames, filename, metadata)
```

Comment:
The CSV that is returned does not have 0's for the absence of data at a given time, but rather is a blank space. This is due to the null that is going into the CSV due to the pruning that happens before creating the CSV. We could either change the CSV metadata to start that a blank space represents absence in the data or we will want to fix this to have 0 show up in the CSVs.

Reply:
I'll make the metadata change unless there are strong preferences for re-populating with a `0` value.